DuckDuckGo.Fluent.Plugin/QrFunctions.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Avalonia.Controls;
+using Avalonia.Media.Imaging;
+using Avalonia.Threading;
+using Blast.API.Core.Processes;
+using Blast.API.Graphics;
+using Blast.API.Processes;
+using Blast.Core.Results;
+using SoftCircuits.HtmlMonkey;
+using static DuckDuckGo.Fluent.Plugin.JsonResult;
+using static DuckDuckGo.Fluent.Plugin.AppFunctions;
+using static DuckDuckGo.Fluent.Plugin.DuckDuckGoSearchResult;
+
+namespace DuckDuckGo.Fluent.Plugin
+{
+    public class QrFunctions
+    {
+        public static async Task<DuckDuckGoSearchResult> GetQrImage(string searchedText)
+        {
+            string url = GetEndpointUrl(searchedText, true);
+            DuckDuckGoApiResult apiResult = await HttpCalls.GetApiResult(url);
+
+            if (apiResult == null) return null;
+
+            const string resultType = "QR";
+
+            if (!apiResult.AnswerType.Equals("qrcode") || string.IsNullOrWhiteSpace(apiResult.Answer)) return null;
+
+            string info = apiResult.Answer;
+
+            BitmapImageResult bitmapImageResult;
+            HtmlDocument document = HtmlDocument.FromHtml(info);
+            IEnumerable<HtmlElementNode> nodes = document.Find("img");
+            HtmlElementNode htmlElementNode = nodes.ElementAt(0);
+
+            if (htmlElementNode.Attributes.Count == 0)
+            {
+                bitmapImageResult = null;
+            }
+            else
+            {
+                HtmlAttributeCollection htmlAttributeCollection = htmlElementNode.Attributes;
+                HtmlAttribute htmlAttribute = htmlAttributeCollection.FirstOrDefault();
+
+                if (htmlAttribute == null) return null;
+
+                string base64String = htmlAttribute.Value;
+                base64String = base64String?[(base64String.IndexOf(",", StringComparison.Ordinal) + 1)..];
+                byte[] imageBytes = Convert.FromBase64String(base64String!);
+                bitmapImageResult = new BitmapImageResult(new MemoryStream(imageBytes));
+            }
+
+            var duckResult = new DuckResult
+            {
+                SearchedText = searchedText, SearchResultType = ResultType.QrCode,
+                SourceUrl = GetGeneralizedUrl("qr code for " + searchedText)
+            };
+
+            return new DuckDuckGoSearchResult(searchedText, searchedText, resultType
+                , QrOperations, 2, duckResult)
+            {
+                Url = searchedText,
+                PreviewImage = bitmapImageResult
+            };
+        }
+
+        public static void SaveQrImage(DuckDuckGoSearchResult duckGoSearchResult)
+        {
+            if (duckGoSearchResult.PreviewImage is { IsEmpty: true }) return;
+
+            Dispatcher.UIThread.Post(() =>
```

Comment:
Instead of using the Avalonia signature, please use FS method `UiUtilities.UiDispatcher.Post(() =>{...})` and pass it the `CancellationToken`